Switch to systemd + other fixes and changes

lvillani opened the pull request:
Switch to systemd + other fixes and changes

mdamiani commented:
@batt Facciamo il passaggio a systemd anche sul progetto cdg? In questo caso ci sono due o tre scriptini da mettere nel formato systemd che partano all'avvio (tipo quello che imposta i gpio, abilitazione del modem umts, etc..).

rasky commented:
@mdamiani per quanto riguarda la parte di rete, l'idea era passare a connman, che gestisce il routing su più interfacce fisiche diverse.

lvillani committed:
Refactor flashsd.sh

This should make it work with out-of-tree buildroot builds and
lays the ground to add the "exploded rootfs to sdcard" mode.

lvillani committed:
Let buildroot manage the rootfs overlay

lvillani committed:
Enable systemd

lvillani committed:
Add and enable dropbear SSH server with socket activation

lvillani committed:
Add and enable bash with colorful prompt

lvillani committed:
barebox: add option to boot from a rw partition on SD card

lvillani committed:
flashsd.sh: add mode to extract rootfs to sdcard

lvillani committed:
Configure journald to store logs in-memory with a circular buffer

lvillani committed:
Enable systemd-networkd

lvillani committed:
Add a couple of aliases

lvillani committed:
Generate dynamic hostname based on eth0's MAC address

This is done only on first boot.

lvillani committed:
Disable unnecessary services

lvillani committed:
Add license header

lvillani committed:
Reformat according to the kernel's C style

lvillani committed:
Move genhostname to a package

lvillani committed:
Add missing return 0
